### PR TITLE
Add Datomic as a project dependency

### PR DESCRIPTION
**Pull request by sduckett**
## Email works needs to save some state
For our [story about logging email events](https://www.pivotaltracker.com/story/show/150309631), this PR adds a dependency on Datomic and provides an initial schema to describe events. You'll notice this is a PR to a topic branch rather than master; once the parts are fully assembled, we'll address production configuration and I'll PR the topic branch.

❓ - After talking with @gws, I wonder if `INITIALIZE_DATOMIC` still necessary configuration, or is it always set `true`?

### CODE REVIEW

**Comment by cap10morgan** [comment]
I'm in favor of getting rid of `INITIALIZE_DATOMIC`. It should always be set to `true`, is idempotent, and was originally created out of an abundance of caution when we were new to Datomic. Most or all of our services have it set to `true` everywhere now and it has never caused a problem.

File: profiles.clj.sample [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{:dev-overrides
```
Comment (sduckett) [inline profiles.clj.sample]: Is this still the favored style of writing profiles?

Comment (cap10morgan) [reply profiles.clj.sample]: Yes. Until leiningen adds a better way to merge profiles, this is what they recommend doing when you want merge behavior (and we do).

File: resources/config.edn [hunk 2]
```diff
@@ -7,4 +7,8 @@
             :kehaar {:incoming-services
                      [{:queue "email-works.email.send"
                        :f email-works.handlers/send-email
-                       :threads 50}]}}}
+                       :threads 50}]}}
+ :datomic {:uri #resource-config/env "EMAIL_WORKS_DATOMIC_URI"
+           :partition :email-works
+           :initialize #resource-config/edn #resource-config/env "INITIALIZE_DATOMIC"
```
Comment (gws) [inline resources/config.edn]: I agree with Wes, let's set it here in the config to `true`, and remove all other instances of `INITIALIZE_DATOMIC` from the codebase.

File: README.md [hunk 3]
```diff
@@ -14,12 +25,20 @@ The RabbitMQ broker connection is configured in `resources/config.edn` in the `[
 
 The default config.edn in the repo allows you to set these parameters via the environment variables `RABBITMQ_PORT_5672_TCP_ADDR` and `RABBITMQ_PORT_5672_TCP_PORT`, which are used by `docker-compose` but can also just be set directly.
 
-### queue
+### Queues
 
-The queue name email-works listens on is configured in `resources/config.edn` in the `[:rabbitmq :queues]` keypath like this:
+The queue names email-works listens on is configured in
```
Comment (gws) [inline README.md]: "The queues `email-works` listens on are configured in"

**Pull request merged**